library/src/main/java/com/owncloud/android/lib/resources/users/CheckRemoteWipeRemoteOperation.java

```diff
@@ -7,65 +7,78 @@
 import com.owncloud.android.lib.common.operations.RemoteOperation;
 import com.owncloud.android.lib.common.operations.RemoteOperationResult;
 import com.owncloud.android.lib.common.utils.Log_OC;
 
 import org.apache.commons.httpclient.HttpStatus;
-import org.apache.commons.httpclient.methods.Utf8PostMethod;
 import org.json.JSONObject;
 
+import java.util.HashMap;
+
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+
 
 /**
  * Remote operation performing check if app token is scheduled for remote wipe
  */
 
-public class CheckRemoteWipeRemoteOperation extends RemoteOperation {
+public class CheckRemoteWipeRemoteOperation extends RemoteOperation<Void> {
 
     private static final String TAG = CheckRemoteWipeRemoteOperation.class.getSimpleName();
-    private static final int SYNC_READ_TIMEOUT = 40000;
-    private static final int SYNC_CONNECTION_TIMEOUT = 5000;
     private static final String REMOTE_WIPE_URL = "/index.php/core/wipe/check";
 
     // JSON node names
     private static final String WIPE = "wipe";
 
+    private String authToken;
+
+    public CheckRemoteWipeRemoteOperation(String authToken) {
+        this.authToken = authToken;
+    }
+
     /**
      * @param client Client object
      */
     @Override
-    protected RemoteOperationResult run(OwnCloudClient client) {
-        Utf8PostMethod postMethod = null;
-        RemoteOperationResult result;
+    public RemoteOperationResult<Void> run(NextcloudClient client) {
+        PostMethod postMethod = null;
+        RemoteOperationResult<Void> result;
 
         try {
-            postMethod = new Utf8PostMethod(client.getBaseUri() + REMOTE_WIPE_URL + JSON_FORMAT);
+            HashMap<String, String> map = new HashMap<>();
+            map.put(REMOTE_WIPE_TOKEN, authToken);
+
+            String jsonString = new Gson().toJson(map);
+
+            RequestBody requestBody = RequestBody.create(MediaType.parse("application/json"), jsonString);
+            postMethod = new PostMethod(client.getBaseUri() + REMOTE_WIPE_URL + JSON_FORMAT, true, requestBody);
             postMethod.addRequestHeader(CONTENT_TYPE, FORM_URLENCODED);
-            postMethod.setParameter(REMOTE_WIPE_TOKEN, client.getCredentials().getAuthToken());
 
-            int status = client.executeMethod(postMethod, SYNC_READ_TIMEOUT, SYNC_CONNECTION_TIMEOUT);
+            int status = client.execute(postMethod);
 
             if (HttpStatus.SC_OK == status) {
                 String response = postMethod.getResponseBodyAsString();
 
                 JSONObject json = new JSONObject(response);
 
                 if (json.getBoolean(WIPE)) {
-                    result = new RemoteOperationResult(true, postMethod);
+                    result = new RemoteOperationResult<>(true, postMethod);
                 } else {
-                    result = new RemoteOperationResult(false, postMethod);
+                    result = new RemoteOperationResult<>(false, postMethod);
```

Comment:
If I'm not mistaken, this if statement can be simplified to the following expression:

```java
result = new RemoteOperationResult<>(json.getBoolean(WIPE), postMethod);
```